core/src/main/java/com/datastax/oss/driver/internal/core/metadata/NodeStateManager.java

```diff
@@ -185,10 +185,10 @@ private void onDebouncedTopologyEvent(TopologyEvent event) {
             }
           } else {
             LOG.debug(
-                "[{}] Received UP event for unknown node {}, adding it",
+                "[{}] Received UP event for unknown node {}, refreshing node list",
                 logPrefix,
                 event.broadcastRpcAddress);
-            metadataManager.addNode(event.broadcastRpcAddress);
+            metadataManager.refreshNodes();
```

Comment:
Note that `refreshNodes` an async operation and we aren't waiting for the result. That's probably fine though, we're in an async code path anyway (debounced handling of topology events).

Reply:
Indeed, but `addNode` is also async. That's fine.